stream/src/main/scala/org/apache/pekko/stream/MapAsyncPartitioned.scala

```diff
@@ -23,102 +23,29 @@ import scala.util.control.{ NoStackTrace, NonFatal }
-        orderedOutput = true,
-        parallelism,
-        extractPartitionWithCtx(extractPartition),
-        fWithCtx[In, T, Ctx, Partition](f)))
-
-  def mapSourceWithContextUnordered[In, Ctx, T, Partition, Mat](flow: SourceWithContext[In, Ctx, Mat],
-      parallelism: Int)(extractPartition: In => Partition)(
-      f: (In, Partition) => Future[T]): SourceWithContext[T, Ctx, Mat] =
-    flow.via(
-      new MapAsyncPartitioned[(In, Ctx), (T, Ctx), Partition](
-        orderedOutput = false,
-        parallelism,
-        extractPartitionWithCtx(extractPartition),
-        fWithCtx[In, T, Ctx, Partition](f)))
-
-  def mapFlowOrdered[In, Out, T, Partition, Mat](flow: Flow[In, Out, Mat], parallelism: Int)(
-      extractPartition: Out => Partition)(
-      f: (Out, Partition) => Future[T]): Flow[In, T, Mat] =
-    flow.via(new MapAsyncPartitioned[Out, T, Partition](orderedOutput = true, parallelism, extractPartition,
-      f))
-
-  def mapFlowUnordered[In, Out, T, Partition, Mat](flow: Flow[In, Out, Mat], parallelism: Int)(
-      extractPartition: Out => Partition)(
-      f: (Out, Partition) => Future[T]): Flow[In, T, Mat] =
-    flow.via(new MapAsyncPartitioned[Out, T, Partition](orderedOutput = false, parallelism,
-      extractPartition, f))
-
-  def mapFlowWithContextOrdered[In, Out, CtxIn, CtxOut, T, Partition, Mat](
-      flow: FlowWithContext[In, CtxIn, Out, CtxOut, Mat], parallelism: Int)(
-      extractPartition: Out => Partition)(
-      f: (Out, Partition) => Future[T]): FlowWithContext[In, CtxIn, T, CtxOut, Mat] =
-    flow.via(
-      new MapAsyncPartitioned[(Out, CtxOut), (T, CtxOut), Partition](
-        orderedOutput = true,
-        parallelism,
-        extractPartitionWithCtx(extractPartition),
-        fWithCtx[Out, T, CtxOut, Partition](f)))
-
-  def mapFlowWithContextUnordered[In, Out, CtxIn, CtxOut, T, Partition, Mat](
-      flow: FlowWithContext[In, CtxIn, Out, CtxOut, Mat], parallelism: Int)(extractPartition: Out => Partition)(
-      f: (Out, Partition) => Future[T]): FlowWithContext[In, CtxIn, T, CtxOut, Mat] =
-    flow.via(
-      new MapAsyncPartitioned[(Out, CtxOut), (T, CtxOut), Partition](
-        orderedOutput = false,
-        parallelism,
-        extractPartitionWithCtx(extractPartition),
-        fWithCtx[Out, T, CtxOut, Partition](f)))
-
-  private[stream] val NotYetThere: Failure[Nothing] = Failure(new Exception with NoStackTrace)
-
-  private[stream] final class Holder[In, Out](
-      val in: In,
-      var out: Try[Out],
-      callback: AsyncCallback[Holder[In, Out]]) extends (Try[Out] => Unit) {
-
-    // To support both fail-fast when the supervision directive is Stop
-    // and not calling the decider multiple times (#23888) we need to cache the decider result and re-use that
-    private var cachedSupervisionDirective: Option[Supervision.Directive] = None
+  private final class Holder[In, Out](val in: In, var out: Try[Out], val cb: AsyncCallback[Holder[In, Out]]) extends (
```

Comment:
could we use the name `callback` instead of `cb`?

Reply:
- I changed it to `cb` as the same in `Holder` in `MapAsync`.
- the `cb` is ok, Cats-Effect use `cb` in unsafeRunAsync too, maybe that's fine to keep it?